WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift

```diff
@@ -0,0 +1,301 @@
+import SwiftUI
+
+/// View to set budget for a new Blaze campaign
+struct BlazeBudgetSettingView: View {
+
+    @Environment(\.dismiss) private var dismiss
+    @State private var showingImpressionInfo = false
+    @State private var showingDurationSetting = false
+
+    @ObservedObject private var viewModel: BlazeBudgetSettingViewModel
+
+    init(viewModel: BlazeBudgetSettingViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(spacing: Layout.sectionSpacing) {
+            // Cancel button
+            HStack {
+                Button(Localization.cancel) {
+                    dismiss()
+                }
+                Spacer()
+            }
+
+            mainContentView
+        }
+        .padding(Layout.contentPadding)
+        .safeAreaInset(edge: .bottom) {
+            footerView
+        }
+        .sheet(isPresented: $showingImpressionInfo) {
+            if #available(iOS 16, *) {
+                impressionInfoView.presentationDetents([.medium, .large])
+            } else {
+                impressionInfoView
+            }
+        }
+        .sheet(isPresented: $showingDurationSetting) {
+            if #available(iOS 16, *) {
+                durationSettingView.presentationDetents([.medium, .large])
```

Comment:
Design discussion:

When the sheet is pulled to show larger detent here, it leaves quite a big empty space that does nothing. This is with an already large font:


https://github.com/woocommerce/woocommerce-ios/assets/266376/1cdbe415-add1-45f5-9ad6-6724956b611c

I wonder if it's needed, or it's fine to leave it this way?

Reply:
Good point! I added a change in abd547607db8e435c190249e6f27331d14de9fed to decide the presentation detent depending on size category. If an accessibility font is being used, large detent is added to prepare for the worst case.